dashboard/app/models/ability.rb

```diff
@@ -423,12 +423,12 @@ def initialize(user)
 
     if user.persisted?
       # These checks control access to Javabuilder.
-      # All verified instructors and students of verified instructors can generate
-      # a Javabuilder session token to run Java code.
+      # All verified instructors and can generate a Javabuilder session token to run Java code.
+      # Students who are also assigned to a CSA section with a verified instructor can run Java code.
       # Verified instructors can access and run Java Lab exemplars.
       # Levelbuilders can access and update Java Lab validation code.
       can :get_access_token, :javabuilder_session do
-        user.verified_instructor? || user.student_of_verified_instructor?
+        user.verified_instructor? || user.sections_as_student.any? {|s| s.assigned_csa? && s.teacher.verified_instructor?}
```

Comment:
Might want to safe navigate to verified_instructor, I think it's possible s.teacher count return nil if a teacher account has been deleted and the section still exists

Reply:
done!

Reply:
I think you updated another line (also seems good) but not this one?

Reply:
whoops good catch